src/pages/tutorial/cw-contract/testing-query.mdx

```diff
@@ -0,0 +1,185 @@
+}
+```
+
+We needed to produce two entities for the `query` functions: the `deps` and `env` instances.
+Fortunately, `cosmwasm-std` provides utilities for testing those -
+[`mock_dependencies`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/testing/fn.mock_dependencies.html)
+and [`mock_env`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/testing/fn.mock_env.html)
+functions.
+
+You may notice the dependencies mock of a type
+[`OwnedDeps`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.OwnedDeps.html) instead of
+`Deps`, which we need here - this is why the
+[`as_ref`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.OwnedDeps.html#method.as_ref)
+function is called on it. If we looked for a `DepsMut` object, we would use
+[`as_mut`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.OwnedDeps.html#method.as_mut)
+instead.
+
+We can rerun the test, and it should still pass. But when we think about that test reflecting the
+actual use case, it is inaccurate. The contract is queried, but it was never instantiated! In
+software engineering, it is equivalent to calling a getter without constructing an object - taking
+it out of nowhere. It is a lousy testing approach. We can do better:
+
+```rust copy filename="src/contract.rs"
+// ...
+
+#[cfg(test)]
+mod tests {
+    use cosmwasm_std::from_json;
+    use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info};
+
+    use super::*;
+
+    #[test]
+    fn greet_query() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+
+        instantiate(
+            deps.as_mut(),
+            env.clone(),
+            mock_info("sender", &[]),
+            Empty {},
+        )
+        .unwrap();
+
+        let resp = query(deps.as_ref(), env, QueryMsg::Greet {}).unwrap();
+        let resp: GreetResp = from_json(&resp).unwrap();
+        assert_eq!(
+            resp,
+            GreetResp {
+                message: "Hello World".to_owned()
+            }
+        );
+    }
+}
+```
+
+A couple of new things here. First, I extracted the `deps` and `env` variables to their variables
+and passed them to calls. The idea is that those variables represent some blockchain persistent
+state, and we don't want to create them for every call. We want any changes to the contract state
+occurring in `instantiate` to be visible in the `query`. Also, we want to control how the
+environment differs on the query and instantiation.
+
+The `info` argument is another story. The message info is unique for each message sent. To create
+the `info` mock, we must pass two arguments to the
+[`mock_info`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/testing/fn.mock_info.html) function.
+
+First is the address performing a call. It may look strange to pass `sender` as an address instead
+of some mysterious `wasm` followed by hash, but it is a valid address. For testing purposes, such
+addresses are typically better, as they are way more verbose in case of failing tests.
```

Comment:
"Verbose" would mean it contains more text. I think it's less :)

Reply:
Readable suits better here